packages/blade/src/components/Button/_decisions/decisions.md

```diff
@@ -0,0 +1,81 @@
+# Button Decisions <!-- omit in toc -->
+- [Design](#design)
+- [Internal BaseButton Component](#internal-basebutton-component)
+- [Button Component](#button-component)
+- [Component Breakdown](#component-breakdown)
+- [Open Questions](#open-questions)
+
+## Design
+[Figma Link](https://www.figma.com/file/jubmQL9Z8V7881ayUD95ps/Blade---Payment-Light?node-id=5200%3A0) to all variants of the Button component
+
+## Internal BaseButton Component
+-  Internal component that exposes certain extra props to enable creation of Component Level Action Buttons like `positive`, `negative`, `neutral`
+-  **BaseButton API**:
+   -  **`variant`**: `primary` | `secondary` | `tertiary` | `link`
+      -  **default**: `primary`
+      -  Link will look like an inline Link component but will be rendered as `button` element
+   -  **`intent`**: `positive` | `negative` | `notice` | `info` | `neutral`
+      -  **default**: `undefined`
+      -  We will use this to set the `intent` for internal Blade components that have positive, negative, notice, info, or neutral buttons like an `Alert` component
+   -  **`contrast`**: `high` | `low`
+      -  **default**: `low`
+      -  We need this for internal Blade components since a `positive` intent could have a `high` as well as a `low` contrast button
+   -  **`size`**: `large` | `medium` | `small` | `xsmall`
+      - **default**: `medium`
+   -  **`children`**: `string`
+   -  **`icon`**: `IconComponent`
+      -  **default**: `undefined`
+   -  **`iconPosition`**: `left` | `right`
+      -  **default**: `right`
+   -  **`isDisabled`**: `true` | `false`
+      -  **default**: `false`
+      -  Should we call this `disabled` instead?
+   -  **`isFullWidth`**: `true` | `false`
+      -  **default**: `false`
+      -  Should we call this `fullWidth` instead?
```

Comment:
If the `isFullWidth` is false, then how are we going to calculate the width of the button? Or will it just take the width depending on the content (icon, label, etc)?

Reply:
isFullWidth will take up the parent container's full width.

Reply:
@sumitd94, yes it will calculate the width based on content & padding.  When we enable `isFullWidth` it will take up the parent's width